### PR TITLE
For #20438 - Display a list of history metadata groups

### LINKED ISSUE #20438: Display a list of history metadata groups

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #20438

Needs https://github.com/mozilla-mobile/android-components/pull/10593/files

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataFeature.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            val historyMetadata = mutableListOf<HistoryMetadataContainer>()
+
+            for ((title, data) in historyMetadataMap) {
+                historyMetadata.add(
+                    HistoryMetadataContainer(
+                        title = title,
+                        historyMetadata = data
+                    )
+                )
+            }
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataFeature.kt:47]: Maybe after the `groupBy` you could do `.map` avoid creating a new list and the looping over the  `historyMetadataMap`

File: app/src/main/res/layout/history_metadata_header.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="56dp"
+    android:layout_marginTop="40dp">
```
Comment (csadilek) [inline app/src/main/res/layout/history_metadata_header.xml:10]: Can we change this to 20dp in all three headers? There's currently a massive gap between all those sections and they are already disappearing below the fold.

Comment (gabrielluong) [reply app/src/main/res/layout/history_metadata_header.xml:10]: Will follow up on this separately after we speak to UX.

Comment (csadilek) [reply app/src/main/res/layout/history_metadata_header.xml:10]: We can bring this up separately and verify.

**Review by csadilek — APPROVED** [review]

**Pull request merged**